Rakefile

```diff
@@ -425,3 +425,288 @@ end
+          # bolt file upload <src> <dest> [options]
+          @pwsh_command[:options] << {
+            name:                       'source',
+            help_msg:                   'The source file or directory to upload',
+            type:                       'string',
+            switch:                     false,
+            mandatory:                  true,
+            position:                   0,
+            ruby_arg:                   'bare',
+            validate_not_null_or_empty: true
+          }
+          @pwsh_command[:options] << {
+            name:                       'destination',
+            type:                       'string',
+            switch:                     false,
+            mandatory:                  true,
+            position:                   1,
+            help_msg:                   'The destination to upload to',
+            ruby_arg:                   'bare',
+            validate_not_null_or_empty: true
+          }
+        when 'secret'
+          # bolt secret encrypt <plaintext> [options]
+          # bolt secret decrypt <ciphertext> [options]
+          @pwsh_command[:options] << {
+            name:                       'text',
+            help_msg:                   "The text to #{action}",
+            type:                       'string',
+            switch:                     false,
+            mandatory:                  true,
+            position:                   0,
+            ruby_arg:                   'bare',
+            validate_not_null_or_empty: true
+          }
+        when 'project'
+          # bolt project init [directory] [options]
+          @pwsh_command[:options] << {
+            name:                       'directory',
+            help_msg:                   'The directory',
+            mandatory:                  false,
+            type:                       'string',
+            switch:                     false,
+            position:                   0,
+            ruby_arg:                   'bare',
+            validate_not_null_or_empty: true
+          }
+        end
+
+        help_text[:flags].reject { |o| o =~ /verbose|debug|help/ }.map do |option|
+          ruby_param = parser.top.long[option]
+          pwsh_name = option.gsub("-", "")
+
+          next if pwsh_name == 'version'
+
+          pwsh_param = {
+            name:       pwsh_name,
+            type:       'string',
+            switch:     false,
+            mandatory:  false,
+            help_msg:   ruby_param.desc.map { |d| d.gsub("<", "&lt;") }.join("\n"),
+            ruby_short: ruby_param.short.first,
+            ruby_long:  ruby_param.long.first,
+            ruby_arg:   ruby_param.arg,
+            ruby_orig:  option
+          }
+
+          case ruby_param.class.to_s
+          when 'OptionParser::Switch::RequiredArgument'
+            # this isn't quite the same thing as a mandatory pwsh parameter
+            # pwsh_param[:mandatory] = true
```

Comment:
Should mandatory be false in that case?

Reply:
In line 636 I set `mandatory:` to false in the intializer, so each param is set to false when created. Also all PowerShell parameters have `Mandatory` set to `$false` by default, so there isn't a need to be explicit in the pwsh code.